### ISSUE TITLE
Issue: High CPU usage when opening the settings page

### ISSUE THREAD

**Issue by Mishasama** [Ephellon/Twitch-Tools#27]
**Describe the bug**

When opening and viewing the settings page, the CPU and memory usage is unusually high.

**To Reproduce**

1. Open the settings
2. See error

**Expected behavior**

A settings page shouldn't have such a high CPU usage, is something caught in a dead loop?

**Screenshots**

![image](https://github.com/user-attachments/assets/5c81e7f3-4c94-4e76-9564-e4753a9e0366)

**Extension Information**
 - Version: `5.33.4.6/7`
 - Source: `store` & `GitHub`
 - Browser: `chrome` & `edge`
 - Operating System: `Windows`

**Additional comments**

CPU usage drops to single digits when placing the page in the background without displaying it, is this caused by any active element rendering over and over again?

**Comment by Ephellon** [comment]
I can get spikes up to `54.6%`. Please provide detailed reproduction steps. If you simply just opened the Settings page, would you be ok with detailing your available RAM (see [`navigator.deviceMemory`](https://developer.mozilla.org/ja/docs/Web/API/Navigator/deviceMemory))?

1. Open Settings
2. Right click and select `Inspect`
3. Select the `Console` tab
4. Type [`navigator.deviceMemory`](https://developer.mozilla.org/ja/docs/Web/API/Navigator/deviceMemory)
5. Let me know what number shows up. Mine for example, is `8`, meaning I have at least `8GB` available in the browser

**Comment by Mishasama** [comment]
8
![image](https://github.com/user-attachments/assets/61e313d5-2ef0-4622-a5a8-8ba3b3e033e4)

**Comment by Ephellon** [comment]
Thank you. I'll continue the stress tests to see what module is causing hiccups

**Comment by Ephellon** [comment]
Found the problem(s), some events were set to fire at `10ms`; changed to `100ms` and 5min later CPU usage on Settings page is at `0.1%`

![image](https://github.com/user-attachments/assets/c56ef730-a1a6-453d-a3d0-0545ee5419d9)

**Comment by Ephellon** [comment]
Found issue(s), some events were set to fire at `10ms`. Changed to `100ms`, and CPU usage is `<1%` 5mins later.

![image](https://github.com/user-attachments/assets/86c8d9db-11be-4983-94a8-d00588ad24cb)

**Comment by Ephellon** [comment]
@Mishasama please try [v5.33.4.8-beta](https://github.com/Ephellon/Twitch-Tools/releases/tag/v5.33.4.8-beta) to see if the Settings page has improvements on your system 😊

[ttv-tools.v5.33.4.8-beta.zip](https://github.com/user-attachments/files/16825815/ttv-tools.zip)

**Comment by Mishasama** [comment]
> Max spike `90%` while typing and moving mouse simultaneously.

Indeed, this happens once these actions are active, even with the .8 version.
Are there any features that use these determinations?

And still get high usage on the stream page.

<hr>

Maybe... try asking ChatGPT to help with debugging?

There are a few clues in my observations: 

1. <s>Are there concentrated polling behaviors? For example, different functions execute in parallel or for a short period.</s> How about 1 second?
2. Are there any incorrect nested loop conditions? This is commonly referred to as an infinite loop.
3. Is there real-time animation rendering with high resource consumption? Such as the high frame rate rainbow animation seen in "Buy When Available." 

<hr>

I found another issue.
 Recover Frames
It will constantly pause and resume playback for short periods.
When we need this function, it means that our network or hardware has a problem that cannot be quickly fixed.

**Comment by Ephellon** [comment]
Stream page, yes, I want to start on the smaller, more manageable Setting spage first to see what can be changed.

----

1. `1 second` won't work, there are dynamic changes being made
2. Only the infinite loops needed are present. The page would freeze if there were any unaccounted for...
3. There's always real-time animation playing... [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame). [The rainbow animation uses the GPU](https://developer.mozilla.org/en-US/docs/Web/Performance/Fundamentals#memory_and_power_usage)

----

`Recover Frames` → Please elaborate in a new issue

**Comment by Mishasama** [comment]
Tabs consume about twice as much CPU when displayed in the foreground as when running in the background, so I suspect it's a rendering-related issue. Maybe you can reduce consumption by lowering the frame rate; there's a power-saving mode option in Chrome for lowering the frame rate globally, so maybe that helps somewhat?
But it's also possible that the browser is limiting the behavior of the tabs in the background, which appears to be decreasingly usage...

**Comment by Ephellon** [comment]
Those settings are managed by Chrome on the Application scale—I have little to no control over that. The best I can do is change how the extension works...

**Comment by Mishasama** [comment]
![image](https://github.com/user-attachments/assets/184ecffb-4ea8-48ad-82b3-deb9f37a108d)
Bad news, still very high.

**Comment by Ephellon** [comment]
May be fixed by [`5.35.1`](https://github.com/Ephellon/Twitch-Tools/releases/tag/5.35.1)

**Comment by Mishasama** [comment]
<img width="700" height="112" alt="Image" src="https://github.com/user-attachments/assets/4a448eee-1069-4152-919b-831705b93252" />

When moving the mouse over the settings page, it is still very high, even though I have already upgraded to 14700K...
